behaviour/graql/language/delete.feature

```diff
@@ -16,12 +33,566 @@
+      | P3   | key   | ref:2     |
+      | FR   | key   | ref:3     |
+
+    Then graql delete
+      """
+      match
+        $r (friend: $x, friend: $y, friend: $z) isa friendship;
+        $x isa person, has ref 0;
+        $y isa person, has ref 1;
+        $z isa person, has ref 2;
+      delete
+        $r (friend: $x);
+        $r (friend: $y);
+      """
+
+    Then get answers of graql query
+      """
+      match $r (friend: $x) isa friendship; get;
+      """
+    Then uniquely identify answer concepts
+      | r     | x    |
+      | FR    | P3   |
+
+
+  Scenario: delete more role players than exist throws
+    When graql insert
+      """
+      insert
+      $x isa person, has ref 0;
+      $y isa person, has ref 1;
+      $r (friend: $x, friend: $y) isa friendship, has ref 2;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | FR   | key   | ref:2     |
+
+    Then graql delete throws
+      """
+      match
+        $x isa person, has ref 0;
+        $y isa person, has ref 1;
+        $r (friend: $x, friend: $y) isa friendship;
+      delete
+        $r (friend: $x, friend: $x);
+      """
+
+
+  Scenario: delete all role players of relation cleans up relation instance
+    When graql insert
+      """
+      insert
+      $x isa person, has ref 0;
+      $y isa person, has ref 1;
+      $r (friend: $x, friend: $y) isa friendship, has ref 2;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | FR   | key   | ref:2     |
+
+    Then graql delete
+      """
+      match
+        $x isa person, has ref 0;
+        $y isa person, has ref 1;
+      delete
+        $x isa person;
+        $y isa person;
+      """
+
+    Then get answers of graql query
+      """
+      match $r isa friendship; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: delete a role player with too-specific (downcasting) role throws
+    Given graql define
+      """
+      define
+      special-friendship sub friendship,
+        relates special-friend as friend;
+      """
+    Given the integrity is validated
+
+    When graql insert
+      """
+      insert
+      $x isa person, has ref 0;
+      $y isa person, has ref 1;
+      $r (friend: $x, friend: $y) isa friendship, has ref 2;
+      """
+    When the integrity is validated
+
+    Then concept identifiers are
+      |      | check | value     |
+      | P1   | key   | ref:0     |
+      | P2   | key   | ref:1     |
+      | FR   | key   | ref:2     |
+
+    Then graql delete throws
+      """
+      match
+        $x isa person, has ref 0;
+        $y isa person, has ref 1;
+        $r (friend: $x, friend: $y) isa friendship;
+      delete
+        $r (special-friend: $x);
+      """
+
```

Comment:
I think we should also have some cases for deleting larger patterns in case something unexpected happens

Reply:
this is actually causing an issue :P